master/buildbot/process/properties.py

```diff
@@ -421,6 +421,32 @@ def getRenderingFor(self, build):
 
 
 @implementer(IRenderable)
+class _SecretRenderer(object):
+
+    def __init__(self, secret_name):
+        self.secret_name = secret_name
+
+    @defer.inlineCallbacks
+    def getRenderingFor(self, build):
+        servicenames = [service.name for service in build.getBuild().master.services]
```

Comment:
does not look right. why get does not work?

try catch is better solution than all the time computing that list of names